jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/bigquery/BigQueryQueryRewriterTest.java

```diff
@@ -0,0 +1,121 @@
+                AdapterProperties.emptyProperties());
+        this.queryRewriter = new BigQueryQueryRewriter(dialect, metadataReader, connectionMock);
+        when(connectionMock.createStatement()).thenReturn(this.mockStatement);
+        when(this.mockResultSet.getMetaData()).thenReturn(this.mockResultSetMetaData);
+        when(this.mockStatement.executeQuery(any())).thenReturn(this.mockResultSet);
+        when(this.mockResultSet.next()).thenReturn(true, false);
+        when(this.mockResultSetMetaData.getColumnCount()).thenReturn(1);
+    }
+
+    @Test
+    void testRewriteWithJdbcConnectionEmptyTable() throws AdapterException, SQLException {
+        when(this.mockResultSet.next()).thenReturn(false);
+        when(this.mockResultSetMetaData.getColumnName(1)).thenReturn("1");
+        assertThat(this.queryRewriter.rewrite(this.statement, this.exaMetadata, AdapterProperties.emptyProperties()),
+                equalTo("SELECT * FROM VALUES (1) WHERE false"));
+    }
+
+    @Test
+    void testRewriteWithBigInt() throws AdapterException, SQLException {
+        when(this.mockResultSetMetaData.getColumnName(1)).thenReturn("bigInt");
+        when(this.mockResultSet.getInt("bigInt")).thenReturn(1);
+        when(this.mockResultSetMetaData.getColumnType(1)).thenReturn(Types.BIGINT);
+        assertThat(this.queryRewriter.rewrite(this.statement, this.exaMetadata, AdapterProperties.emptyProperties()),
+                equalTo("SELECT * FROM VALUES (1)"));
+    }
+
+    @CsvSource({ "float, 8, 105.0", //
+            "float, 8, 99.4", //
+            "numeric, 2, 22222.2222", //
+            "numeric, 2, 11.5" })
+    @ParameterizedTest
+    void testRewriteWithFloatingValues(final String columnName, final int type, final double columnValue)
+            throws AdapterException, SQLException {
+        when(this.mockResultSetMetaData.getColumnName(1)).thenReturn(columnName);
+        when(this.mockResultSet.getDouble(columnName)).thenReturn(columnValue);
+        when(this.mockResultSetMetaData.getColumnType(1)).thenReturn(type);
+        assertThat(this.queryRewriter.rewrite(this.statement, this.exaMetadata, AdapterProperties.emptyProperties()),
+                equalTo("SELECT * FROM VALUES (" + columnValue + ")"));
+    }
+
+    @Test
+    void testRewriteWithBoolean() throws AdapterException, SQLException {
+        when(this.mockResultSetMetaData.getColumnName(1)).thenReturn("boolean");
+        when(this.mockResultSet.getBoolean("boolean")).thenReturn(true);
+        when(this.mockResultSetMetaData.getColumnType(1)).thenReturn(Types.BOOLEAN);
+        assertThat(this.queryRewriter.rewrite(this.statement, this.exaMetadata, AdapterProperties.emptyProperties()),
+                equalTo("SELECT * FROM VALUES (true)"));
+    }
+
+    @CsvSource({ "string, 12, hello", //
+            "bytes, -3 ,D7A0FA33B3CDDF0EC29FC989CAB9AB512658F6E9A631FE5008610CBA24A64A5C", //
```

Comment:
Time formats identical to Exasol's? See comment in implementation class above.

Reply:
done